src/app/store/features/creditCardSlice.ts

```diff
@@ -0,0 +1,58 @@
+import { createSlice } from '@reduxjs/toolkit';
+import {
+  CreditCardData,
+  deleteCreditCardFulfiled,
+  getCreditCardsFulfiled,
+} from 'infrastructure/services/creditCard/CreditCardService';
+
+const initialState: CreditCardState = {
+  creditCards: [],
+  defaultCreditCard: {
+    id: '',
+    name: '',
+    time: '',
+    billingInfo: {
+      city: '',
+      name: '',
+      line1: '',
+      line2: '',
+      country: '',
+      district: '',
+      postalCode: '',
+    },
+    depositVerificationStatus: 'notStarted',
+    depositVerificationErrorCode: 'no_error',
+    status: 'pending',
+    data: {
+      mask: '',
+    },
+    errorCode: null,
+  },
+};
+
+const creditCardSlice = createSlice({
+  name: 'creditCard',
+  initialState,
+  reducers: {
+    setDefaultCreditCard: (state, { payload }) => {
+      state.defaultCreditCard = payload;
+    },
+  },
+  extraReducers: builder => {
+    builder.addMatcher(getCreditCardsFulfiled, (state, { payload }) => {
+      state.creditCards = [...payload];
+      state.defaultCreditCard = { ...payload[0] };
```

Comment:
Would be a problem is the first cc from the array is rejected?

Reply:
There is no problem to set a default credit card as rejected. In this case we block the user's deposits and that's it.